sycl/include/sycl/info/ext_intel_device_traits.def

```diff
@@ -2,20 +2,20 @@
 #define __SYCL_PARAM_TRAITS_TEMPLATE_SPEC_NEEDS_UNDEF
 #define __SYCL_PARAM_TRAITS_TEMPLATE_SPEC __SYCL_PARAM_TRAITS_SPEC
 #endif
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, device_id, pi_uint32, PI_DEVICE_INFO_DEVICE_ID)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, pci_address, std::string, PI_DEVICE_INFO_PCI_ADDRESS)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, gpu_eu_count, pi_uint32, PI_DEVICE_INFO_GPU_EU_COUNT)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, gpu_eu_simd_width, pi_uint32, PI_DEVICE_INFO_GPU_EU_SIMD_WIDTH)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, gpu_slices, pi_uint32, PI_DEVICE_INFO_GPU_SLICES)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, gpu_subslices_per_slice, pi_uint32, PI_DEVICE_INFO_GPU_SUBSLICES_PER_SLICE)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, gpu_eu_count_per_subslice, pi_uint32, PI_DEVICE_INFO_GPU_EU_COUNT_PER_SUBSLICE)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, gpu_hw_threads_per_eu, pi_uint32, PI_DEVICE_INFO_GPU_HW_THREADS_PER_EU)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, max_mem_bandwidth, pi_uint64, PI_DEVICE_INFO_MAX_MEM_BANDWIDTH)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, uuid, detail::uuid_type, PI_DEVICE_INFO_UUID)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, free_memory, pi_uint64, PI_EXT_INTEL_DEVICE_INFO_FREE_MEMORY)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, memory_clock_rate, pi_uint32, PI_EXT_INTEL_DEVICE_INFO_MEMORY_CLOCK_RATE)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, memory_bus_width, pi_uint32, PI_EXT_INTEL_DEVICE_INFO_MEMORY_BUS_WIDTH)
-__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, max_compute_queue_indices, pi_int32, PI_EXT_INTEL_DEVICE_INFO_MAX_COMPUTE_QUEUE_INDICES)
+__SYCL_PARAM_TRAITS_SPEC(ext::intel, device, device_id, uint32_t, UR_DEVICE_INFO_DEVICE_ID)
```

Comment:
It probably won't help to reduce amount of changed lines, but may help to reduce cognitive load on reviewers: replacing `pi_*` integer typedefs with actual types could be outlined into a separate PR. 

Such PR should be very easy to review despite the potential changes, because they all are purely mechanical and we can be sure that the change is correct, because we are simply dropping a typedef in favor of an actual type it points to.